js/battledata.js

```diff
@@ -826,11 +826,66 @@ var Tools = {
 			}
 			return {tagName: tagName, attribs: attribs};
 		};
+		var localizeTimes = function (tree) {
+			var times = tree.getElementsByTagName('time');
+			for (var i = 0; i < times.length; i++) {
+				var timeElement = times[i];
+				var time = timeElement.textContent;
+
+				// Require ISO 8601 time. While more time formats are supported by
+				// most JavaScript implementations, it isn't required, and how to
+				// exactly enforce ignoring user agent timezone setting is not obvious.
+				// As dates come from the server which isn't aware of client timezone,
+				// a particular timezone is required.
+				//
+				// This regular expression is split into three groups.
+				//
+				// Group 1 - date
+				// Group 2 - time (seconds and milliseconds are optional)
+				// Group 3 - optional timezone
+				//
+				// Group 1 and group 2 are split to allow using space as a separator
+				// instead of T. Stricly speaking ECMAScript 5 specification only
+				// allows T, however it's more practical to also allow spaces.
+				var parts = /^\s*([+-]?\d{4,}-\d{2}-\d{2})[T ](\d{2}:\d{2}(?::\d{2}(?:\.\d{3})?)?)(Z|[+-]\d{2}:\d{2})?\s*$/i.exec(time);
+				if (!parts) continue;
+
+				var parsedTime = new Date(parts[1] + 'T' + parts[2] + (parts[3] || 'Z').toUpperCase());
+				// Very old (pre-ES5) web browsers may be incapable of parsing ISO 8601
+				// dates. In such a case, gracefully continue without replacing the date
+				// format.
+				if (!parsedTime.getTime()) continue;
+
+				var formattedTime;
+				// Try using Intl API if it exists
+				if (window.Intl && window.Intl.DateTimeFormat) {
+					formattedTime = new Intl.DateTimeFormat(undefined, {month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric'}).format(parsedTime);
+				} else {
+					// toLocaleString even exists in ECMAScript 1, so no need to check
+					// if it exists.
+					formattedTime = parsedTime.toLocaleString();
+				}
+				timeElement.textContent = formattedTime;
+			}
+		};
 		return function (input) {
-			return html.sanitizeWithPolicy(getString(input), tagPolicy);
+			var sanitized = html.sanitizeWithPolicy(getString(input), tagPolicy);
+			// Unfortunately Caja doesn't support modifying tags in case of <time> tag,
+			// so Caja's output is converted into DOM just to modify it. Element name
```

Comment:
I would prefer to avoid accidentally introducing security issues by doing transformations after Caja processes. Doing regex replace before Caja will cause issues, in fact security ones allowing for stuff like room intros tracking user's locale and timezone. Doing regex replace after Caja is possible, albeit on risky side, considering it involves changing safe HTML in weird ways. Probably safe, but I'm not willing to risk it myself.

The reason why I directly work on DOM is that this is very unlikely to cause security issues, as I don't use `innerHTML` in DOM code, instead operate directly on trees.